internal/controller/pkg/revision/establisher.go

```diff
@@ -146,15 +146,38 @@ func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, p
 }
 
 func (e *APIEstablisher) create(ctx context.Context, obj resource.Object, parent resource.Object, opts ...client.CreateOption) error {
-	ref := meta.AsController(meta.TypedReferenceTo(parent, parent.GetObjectKind().GroupVersionKind()))
+	// We add the parent as `owner` of the resources so that the resource doesn't
+	// get deleted when the new revision doesn't include it in order not to lose
+	// user data, such as custom resources of an old CRD.
+	falseVal := false
+	refs := make([]metav1.OwnerReference, len(parent.GetOwnerReferences())+1)
+	for i, ownerRef := range parent.GetOwnerReferences() {
```

Comment:
@muvaf do you think we should copy all owner refs instead of just the parent package? It can be identified via the `pkg.crossplane.io/package` [label we set on the revision](https://github.com/crossplane/crossplane/blob/2215d0bf53671c2477addee3072b6e59f7c57653/internal/controller/pkg/manager/reconciler.go#L337).

Reply:
Done!